instrumentation/xxl-job/xxl-job-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/xxljob/common/XxlJobSpanNameExtractor.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.xxljob.common;
+
+import static io.opentelemetry.javaagent.instrumentation.xxljob.common.XxlJobConstants.SCRIPT_JOB_TYPE;
+
+import com.xxl.job.core.glue.GlueTypeEnum;
+import io.opentelemetry.instrumentation.api.incubator.semconv.code.CodeAttributesGetter;
+import io.opentelemetry.instrumentation.api.incubator.semconv.code.CodeSpanNameExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+
+class XxlJobSpanNameExtractor implements SpanNameExtractor<XxlJobProcessRequest> {
+  private final SpanNameExtractor<XxlJobProcessRequest> codeSpanNameExtractor;
+
+  XxlJobSpanNameExtractor(CodeAttributesGetter<XxlJobProcessRequest> getter) {
+    codeSpanNameExtractor = CodeSpanNameExtractor.create(getter);
+  }
+
+  @Override
+  public String extract(XxlJobProcessRequest request) {
+    GlueTypeEnum glueType = request.getGlueType();
+    if (SCRIPT_JOB_TYPE.contains(glueType.getDesc())) {
```

Comment:
perhaps it would be better to use `GlueTypeEnum.isScript` https://github.com/xuxueli/xxl-job/blob/77441a0e41e16ae3a65b232512f44eb60f0a74af/xxl-job-core/src/main/java/com/xxl/job/core/glue/GlueTypeEnum.java#L32